odl/solvers/advanced/split_bregman.py

```diff
@@ -0,0 +1,153 @@
+# Copyright 2014-2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""The split bregman method for L1 regularized problems.
+
+The method is proposed in `[1]_` and used to solve TV style regularized
+problems.
+
+.. [1]: ftp://ftp.math.ucla.edu/pub/camreport/cam08-29.pdf
+"""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+from builtins import super
+
+# External
+
+# Internal
+from odl.operator.operator import Operator
+from odl.solvers.iterative import conjugate_gradient
+from odl.util import pspace_squared_sum
+
+__all__ = ('l2_gradient', 'split_bregman_solver',)
+
+
+class l2_gradient(Operator):
+
+    """ Gradient of ``||op(x) - rhs||_2^2``"""
+
+    def __init__(self, op, rhs):
+        """Initialize an instance.
+
+        Parameters
+        ----------
+        H_grad : linear `Operator`
+            Forward operator
+        rhs : ``op.range`` element
+            Right hand side of the inverse problem.
+
+        """
+        self.op = op
+        self.rhs = rhs
+
+        assert rhs in op.domain
+        assert op.is_linear
+
+        super().__init__(op.domain, op.domain)
+
+    def _call(self, x, out=None):
+        return self.op.adjoint(self.op(x) - self.rhs, out=out)
+
+    def derivative(self, x):
+        return self.op.adjoint * self.op
+
+
+def split_bregman_solver(H_grad, Phi, x, lam,
+                         niter=1, inner_niter=1,
+                         isotropic=False, inner_solver=None,
+                         partial=None):
+    """ Reconstruct with split Bregman.
+
+    Solves the L1 regularized problem
+
+        ``min_x H(x) + mu ||Phi(x)||_1``
```

Comment:
* Change `mu` to `lam`
* Is this the formulation we want to solve, do we want to be more general?